### ISSUE TITLE
Crash on Multitask (Huawei)

### ISSUE THREAD

**Issue by SteffoSpieler** [stefan-niedermann/nextcloud-deck#1025]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug**

When opening the app and then adding an app to the multitask view (and resizing it) the app will crash.

1. Open another app in a window.
2. Open Deck.
3. Drag the windowed app to the Multitask view.
4. Resize both apps.
5. See the crash.

**Expected behavior**

It should not crash.

**Versions**
 - Nextcloud: 20.0.10
 - Nextcloud Deck: 1.2.7
 - Nextcloud Android: 3.16.1
 - Nextcloud Android Deck: 1.18.3 (1018003)

**Smartphone (please complete the following information):**
- Device: Huawei P30 Pro
- Android-Version: 10 (EMUI 11)
- App-Store:
  - [ ] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [x] F-Droid
  - [ ] Huawei AppGallery

**Stacktrace**
```
Full Crash:

App Version: 1.18.3
App Version Code: 1018003
App Flavor: fdroid

Files App Version Code: 30160190

---

OS Version: 4.14.116(10.1.0.150C431)
OS API Level: 29
Device: HWVOG
Manufacturer: HUAWEI
Model (and Product): VOG-L29 (VOG-L29EEA)

---

java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.View it.niedermann.android.crosstabdnd.DraggedItemLocalState.getDraggedView()' on a null object reference
	at it.niedermann.android.crosstabdnd.CrossTabDragAndDrop.lambda$register$0$CrossTabDragAndDrop(CrossTabDragAndDrop.java:54)
	at it.niedermann.android.crosstabdnd.-$$Lambda$CrossTabDragAndDrop$sEkzWD_6cDJNiPr4BP6GJbm3lTw.onDrag(Unknown Source:10)
	at android.view.View.callDragEventHandler(View.java:26153)
	at android.view.View.dispatchDragEvent(View.java:26144)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1751)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewGroup.notifyChildOfDragStart(ViewGroup.java:1897)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1744)
	at android.view.ViewRootImpl.handleDragEvent(ViewRootImpl.java:7580)
	at android.view.ViewRootImpl.access$1300(ViewRootImpl.java:165)
	at android.view.ViewRootImpl$ViewRootHandler.handleMessage(ViewRootImpl.java:5253)
	at android.os.Handler.dispatchMessage(Handler.java:110)
	at android.os.Looper.loop(Looper.java:219)
	at android.app.ActivityThread.main(ActivityThread.java:8393)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)

```

**Comment by desperateCoder** [comment]
Hi @SteffoSpieler, thank you for your report.

This is kind of weird, because the code fragment mentioned in the crash report belongs to the drag and drop feature to move cards around. Did it really crash without interaction with the app itself? 

If so, maybe a simple nullcheck would help @stefan-niedermann ?

@SteffoSpieler we'll have a look at this, as soon as we got feedback regarding my question, thanks again for reporting!

**Comment by SteffoSpieler** [comment]
@desperateCoder I made a recording of the bug. This might help more. https://user-images.githubusercontent.com/41108856/122681323-f570ca80-d1f3-11eb-88c4-67bd536ea6c3.mp4

**Comment by desperateCoder** [comment]
@SteffoSpieler oh that's awesome buddy, we'll have a look!

PS: Nice website you got there! I can tell you'll rock your life, as you already achieved quite something already at your young age. Keep it up!
![8f977e058ed7152833ff83eaa5be50a1](https://user-images.githubusercontent.com/6911132/122681578-32898c80-d1f5-11eb-9041-6a4bcf04a533.png)

**Comment by stefan-niedermann** [comment]
> If so, maybe a simple nullcheck would help 

Well this is only a symptom, not the cause of the error. The fragment shouldn't be null at this point. I am not a fan of just hiding the error as we won't remember it and have a bug for a long time then...

I will try to reproduce the issue and maybe we find a better solution, but yeah, if we aren't able to find the root cause, we might at least be able to prevent the crash...

Thanks fir the report @SteffoSpieler 🙂

**Comment by stefan-niedermann** [comment]
@SteffoSpieler since neither @desperateCoder nor me can reproduce this on our physical devices (and also not in emulators), we need some more information:
- Are you using a custom ROM?
- Are you using another launcher than the default launcher?
- Are you able to reliable reproduce this issue *each* time you are performing the steps to reproduce?

**Comment by SteffoSpieler** [comment]
- I am not using a custom ROM.
- I am using the default EMUI launcher.
- I can reproduce this issue each time I am performing the steps to reproduce. (which you can also see in the video)

My Versions:
**Build-Number**: 11.0.0.153
**EMUI-Version**: 11.0.0
**Android-Version**: 10

**Comment by stefan-niedermann** [comment]
Could you please try [this build](https://github.com/stefan-niedermann/nextcloud-deck/suites/3214887669/artifacts/74394515)? You can safely install and use it parallel to the published version.

**Comment by SteffoSpieler** [comment]
The Bug is gone in your build.

**Comment by stefan-niedermann** [comment]
Awesome. Thanks for testing!

We'll ship this fix with `1.19.1`.
